cmake/External_GTest.cmake

```diff
@@ -0,0 +1,58 @@
+find_package(Threads REQUIRED)
+
+include(ExternalProject)
```

Comment:
Having been burned by scripts venturing into dependency management land, the question I've come to ask myself in such cases: who does the project want to cater to? Users who don't necessarily know how to obtain dependencies or users who are already using one (Vcpkg, Conan, etc.) and actually already have things built and a simple `find_package(GTest)` would simply succeed?

I recently engaged in an enlightening [dicsussion](https://discourse.cmake.org/t/depending-on-package-config-file-as-external-project/1501/10?u=mathiasmagnus) around this topic. I myself am deriving some collection of best practices, but CMake is the wild west and it's hard to select the best approach for any given project (requires extensive knowledge).

The AMD libraries still haven't decided on proper defaults and overrides your GTest installs without question. Every configuration time is aggravated by full builds of GTest and Google Bench, even if you have the best and latest installed. Once you start testing their availability, or adding "all or nothing" or specific self-hosting capabilities, complexity starts exploding, hence I'm usually against built-in dependency management when having a say. It's nice to cater to novice users, but don't screw advanced users over.

Reply:
An alternative would be to try `find_package(GTest)` and download/build gtest only when it is not available. This might be the best of both worlds at the cost of some additional logic in the CMake file.

Reply:
Note, how rocPRIM is doing a similar approach (but it actually invokes the `install` target of deps inside [`Dependencies.cmake`](https://github.com/ROCmSoftwarePlatform/rocPRIM/blob/develop/cmake/Dependencies.cmake) which I'd gladly burn to the ground. It's code that shouldn't exist.

If going down this road, I'd minimally suggest:
- Something like `option(FORCE_DOWNLOAD_EXTERNAL)` which has some default.
- If the option is set, do the thing in the scripts currently
- If not, first try a `find_pacakge(GTest)` without `REQUIRED`
  - If present, merrily use it
  - If not, the existing machinery kicks in
- I'd invoke the deps `install` target at configure time, then I don't have to extract the essence of the Package Config file which you do now.
- My own `install` target would include copying over the deps installed dir from my build tree.

This is a fairly comprehensive, self-contained solution that is quite far from zero code, hence why I'd just point people in the direction of using Vcpkg and live a happy life. :)

Reply:
@MathiasMagnus 
> who does the project want to cater to?

Any tool based on or consuming compiled module or header units should be able to use this; any compiler processing C++, on any platform with C++ compilers.